cypress/e2e/Filters/NoTrials/noTrials.js

```diff
@@ -0,0 +1,144 @@
+
+Then('Location by ZIP Code has {int} entered', (value) => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').find('input[type="text"]').should('have.value', value.toString());
+});
+
+Then('Location by ZIP Code has {string} entered', (value) => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').find('input[type="text"]').should('have.value', value);
+});
+
+Then('Radius has {int} miles selected', (value) => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').find('select.usa-select').should('have.value', value.toString());
+});
+
+// Disabled filter states
+Then('Age parameter text input empty and disabled', () => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').find('input[type="number"]').should('be.disabled').and('have.value', '');
+});
+
+Then('Location by ZIP Code text input is empty and disabled', () => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').find('input[type="text"]').should('be.disabled').and('have.value', '');
+});
+
+Then('Radius displays {string} and is disabled', () => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').find('select.usa-select').should('be.disabled');
+	cy.get('.ctla-sidebar').find('select.usa-select option:selected').should('have.text', 'Select');
+});
+
+// Redirection checks with improved error handling
+Then('page redirects to {string}', (urlWithParams) => {
+	// Wait for network calls to complete
+	cy.wait(2000);
+
+	// Add longer timeout for redirection
+	cy.url({ timeout: 10000 }).should('include', urlWithParams.split('?')[0]); // Check base URL
+
+	// If the URL has parameters, check them too
+	if (urlWithParams.includes('?')) {
+		// Extract and check each parameter
+		const params = urlWithParams.split('?')[1].split('&');
+		params.forEach((param) => {
+			cy.url().should('include', param);
+		});
+	}
+
+	// Add assertion retry in case of network delays
+	cy.wrap(null, { timeout: 5000 }).should(() => {
+		cy.url().then((url) => {
+			expect(url).to.include(urlWithParams.split('?')[0]);
+		});
+	});
+});
+
+// Click steps for compatibility with existing tests
+When('user clicks the {string} button', (buttonText) => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').contains('button', buttonText).click();
+});
+
+// Alternative syntax for backward compatibility
+When('clicks the {string} button', (buttonText) => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('.ctla-sidebar').contains('button', buttonText).click();
+});
+
+// CIS Banner check
+Then('CIS banner image displays below the No Trials Found message', () => {
+	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
+	cy.get('[alt="Questions? Chat with an information specialist"]').should('be.visible');
```

Comment:
its better to pass the message as  string parameter so you dont have it hardcoded

Reply:
Then('CIS banner image displays below the No Trials Found message as {string}', (message) => {
	cy.get('.nci-spinner').should('not.exist', { timeout: 10000 });
	cy.get(`[alt="${message}"]`).should('be.visible');
});